configuration/configuration_test.go

```diff
@@ -0,0 +1,831 @@
+	tEnvironmentVariableValueConfigFile = "../config.yaml"
+	tInvalidConfigFilePath              = "../invalid_config.yaml"
+	tTestConfigFilePath                 = "../test/data/multiple_config.yaml"
+
+	// The following will be used to set env variables for testing.
+	tTestEnvString         = "ALMIGHTH_CONFIG_VALUE"
+	tTestEnvInt64    int64 = 12345
+	tTestEnvInt      int   = 12
+	tTestEnvDuration       = "1s"
+	tTestEnvBool           = false
+)
+
+func TestGetDefaultConfigurationFile(t *testing.T) {
+	t.Parallel()
+	assert.Equal(t, configuration.GetDefaultConfigurationFile(), tDefaultConfigurationFile)
+}
+
+func TestGetConfigurationDataSucess(t *testing.T) {
+	t.Parallel()
+	// this, in reality gets set in main
+	os.Setenv(tEnvironmentVariableNameConfigFile, tEnvironmentVariableValueConfigFile)
+	cd, err := configuration.GetConfigurationData()
+	assert.Nil(t, err)
+	assert.NotNil(t, cd)
+}
+
+func TestNewConfigurationDataSuccess(t *testing.T) {
+	t.Parallel()
+	resource.Require(t, resource.UnitTest)
+	configFilePath := tEnvironmentVariableValueConfigFile
+	cd, err := configuration.NewConfigurationData(configFilePath)
+	assert.Nil(t, err)
+	assert.NotNil(t, cd)
+
+}
+
+func TestNewConfigurationDataFail(t *testing.T) {
+	t.Parallel()
+	resource.Require(t, resource.UnitTest)
+	configFilePath := tInvalidConfigFilePath
+	_, err := configuration.NewConfigurationData(configFilePath)
+	assert.NotNil(t, err)
+
+}
+
+func TestGetKeycloakEndpointToken(t *testing.T) {
+	t.Parallel()
+
+	cd := getConfigurationDataHandler()
+	require.NotNil(t, cd)
+
+	viperValue := cd.GetKeycloakEndpointToken()
+
+	testConfigFileMap, err := getConfigFileMap("")
+	require.Nil(t, err)
+	expectedValue := testConfigFileMap.KeycloakEndpointToken
+
+	assert.Equal(t, expectedValue, viperValue)
+
+}
+
+func TestGetPostgresHost(t *testing.T) {
+	t.Parallel()
+
+	envKey := generateEnvKey(varPostgresHost)
+
+	realEnvValue := os.Getenv(envKey) // could be "" as well.
+
+	os.Unsetenv(envKey)
+	defer os.Setenv(envKey, realEnvValue) // set it back before we leave.
+
+	// env variable NOT set, so we check with config.yaml's value
+
+	configurationData := getConfigurationDataHandler()
+	require.NotNil(t, configurationData)
+
+	viperValue := configurationData.GetPostgresHost()
+
+	// now check with the config yaml file.
+	testConfigFileMap, err := getConfigFileMap(tEnvironmentVariableValueConfigFile)
+	require.Nil(t, err)
+	expectedValue := testConfigFileMap.PostgresHost
+
+	assert.NotNil(t, viperValue)
+	assert.Equal(t, expectedValue, viperValue)
+
+	// env variable will now be SET, so now we check with env variable and NOT With config.yaml
+	os.Setenv(envKey, tTestEnvString)
+
+	configurationData = getConfigurationDataHandler()
+	require.NotNil(t, configurationData)
+
+	require.Nil(t, err)
+
+	viperValue = configurationData.GetPostgresHost()
+	assert.Equal(t, tTestEnvString, viperValue)
+
+}
+
+func TestGetPostgresPort(t *testing.T) {
```

Comment:
Almost 30 lines for just testing the port value is way to much IMHO. Especially since there's a lot of repetitive code in each test. If you need to run a setup each test, try using test suites.